ui/hooks/src/use-global-login.ts

```diff
@@ -17,72 +18,92 @@ export interface UseGlobalLoginProps {
   onError?: OnErrorHandler;
   waitForAuth?: (data: boolean) => void;
   onReady?: (data: { ethAddress: string; pubKey: string }) => void;
+  onLoadFromCache?: (data: CurrentUser) => void;
 }
 
 const useGlobalLogin = (props: UseGlobalLoginProps): void => {
-  const { onError } = props;
+  const { onError, onLogin, waitForAuth, onReady, onLogout, onLoadFromCache } = props;
+  const onErrorHandler = React.useRef(onError);
+  const onLoginHandler = React.useRef(onLogin);
+  const waitForAuthHandler = React.useRef(waitForAuth);
+  const onLogoutHandler = React.useRef(onLogout);
+  const onReadyHandler = React.useRef(onReady);
+  const onLoadFromCacheHandler = React.useRef(onLoadFromCache);
 
-  const sdk = getSDK();
+  const sdk = React.useRef(getSDK());
```

Comment:
this is nice, it will get cleaned up when the component that uses it gets unmounted?